vta/hardware/chisel/src/main/scala/core/EventCounters.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package vta.core
+
+import chisel3._
+import chisel3.util._
+import vta.util.config._
+import vta.shell._
+
+/** EventCounters.
+  *
+  * This unit contains all the event counting logic. One common event tracked in
+  * hardware is the number of clock cycles taken to achieve certain task. We
+  * can count the total number of clock cycles spent in a VTA run by checking
+  * launch and finish signals.
+  *
+  * The event counter value is passed to the VCR module via the ecnt port, so
```

Comment:
Can we talk a little bit about how we add a new counter, maybe pointers to the places that need to be changed? it looks like the event count vector has a length set somewhere?

Reply:
Sure, I can add more. The ecnt vector params are taken from [here](https://github.com/dmlc/tvm/blob/be8354b72bac233e6a18c4eefdd4269e72ef017a/vta/hardware/chisel/src/main/scala/core/EventCounters.scala#L38)

see Line 38 from link above

```
  val vp = p(ShellKey).vcrParams
```

The design patter around params is that every time you see something like `myModuleParams" somewhere, then you should go to `myModule` to see what parameters are defined there.

The idea here is that if the number of event counters or `nECnt` is changed in VCR then the interface will automatically change as well, without needing to do it manually. The `EventCounters.scala` file will just implement "what you want to count" logic.

Reply:
Also, I just added a little bit of more info on how to add counters.